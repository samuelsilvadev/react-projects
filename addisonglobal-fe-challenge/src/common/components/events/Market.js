import React from 'react';
import PropTypes from 'prop-types';

import './Market.scss';

const renderButton = (selection) => (
	<button key={selection.id} type="button">
		{selection.name}
	</button>
);

function Market({ name, selections }) {
	return (
		<section>
			<h4 className="market__title">{name}</h4>
			{selections.map(renderButton)}
		</section>
	);
}

Market.propTypes = {
	name: PropTypes.string.isRequired,
	selections: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string,
			name: PropTypes.string,
			price: PropTypes.number
		})
	)
};

Market.defaultProps = {
	selections: []
};

export default Market;
