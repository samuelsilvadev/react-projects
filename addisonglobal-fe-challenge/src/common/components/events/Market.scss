.market {
	&__title {
		font-family: 'Lato', sans-serif;
		font-size: 17px;
		font-weight: 400;
		color: #8190a5;
	}
}
