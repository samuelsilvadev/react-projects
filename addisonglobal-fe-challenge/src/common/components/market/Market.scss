@import '../../scss/main.scss';

.market {
	&__title {
		color: $fourth;
		font-family: 'Lato', sans-serif;
		font-size: 17px;
		font-weight: 400;
		margin-bottom: 10px;
		width: 100%;
	}

	&__selections-wrapper {
		display: flex;
		justify-content: space-between;
	}

	&__btn {
		align-items: center;
		border: 1px solid $fourth;
		border-radius: 5px;
		background: transparent;
		cursor: pointer;
		display: flex;
		flex-direction: column;
		flex-basis: 117.33px;
		flex-shrink: 1;
		padding: 8px;
		transition-property: color, background-color;
		transition-duration: 200ms;

		&--active {
			background: $limegreen;
			color: white;
		}
	}
}
