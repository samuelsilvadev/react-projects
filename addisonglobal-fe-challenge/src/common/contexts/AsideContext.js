import React, { useState, useContext } from 'react';

const AsideContext = React.createContext();

export const AsideContextProvider = (props) => {
	const [isAsideVisible, setAsideVisible] = useState(false);

	const handleOpen = () => {
		setAsideVisible(true);
	};

	const handleClose = () => {
		setAsideVisible(false);
	};

	const context = {
		isAsideVisible,
		handleOpen,
		handleClose
	};

	return <AsideContext.Provider {...props} value={context} />;
};

export const useAsideContext = () => {
	return useContext(AsideContext);
};

export function withAsideContext(Component) {
	function WithAsideContext(props) {
		return (
			<AsideContextProvider>
				<Component {...props} />
			</AsideContextProvider>
		);
	}

	return WithAsideContext;
}
