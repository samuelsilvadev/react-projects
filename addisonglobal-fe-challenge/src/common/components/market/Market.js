import React from 'react';
import PropTypes from 'prop-types';

import './Market.scss';

const renderButton = (selection) => (
	<button className="market__btn" key={selection.id} type="button">
		<span>{selection.name}</span>
		<span>{selection.price}</span>
	</button>
);

function Market({ name, selections }) {
	return (
		<section className="market">
			<h4 className="market__title">{name}</h4>
			<div className="market__selections-wrapper">
				{selections.map(renderButton)}
			</div>
		</section>
	);
}

Market.propTypes = {
	name: PropTypes.string.isRequired,
	selections: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string,
			name: PropTypes.string,
			price: PropTypes.number
		})
	)
};

Market.defaultProps = {
	selections: []
};

export default Market;
