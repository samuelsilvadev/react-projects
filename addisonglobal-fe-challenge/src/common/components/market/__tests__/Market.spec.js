import React from 'react';
import { render } from '@testing-library/react';

import Market from '../Market';

import { AsideContextProvider } from '../../../contexts/AsideContext';

describe('<Market />', () => {
	it('should render correctly with all props', () => {
		const props = {
			name: 'Player to Score First',
			selections: [
				{
					id: '1',
					name: 'Messi',
					price: 1.3
				}
			]
		};
		const { container, queryByText } = render(
			<AsideContextProvider>
				<Market {...props} />
			</AsideContextProvider>
		);

		expect(queryByText(props.name)).toBeVisible();
		expect(queryByText(props.selections[0].name)).toBeVisible();
		expect(queryByText(String(props.selections[0].price))).toBeVisible();
		expect(container.firstChild).toMatchSnapshot();
	});
});
