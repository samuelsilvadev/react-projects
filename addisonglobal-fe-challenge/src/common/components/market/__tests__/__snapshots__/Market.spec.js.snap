// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Market /> should render correctly with all props 1`] = `
<section
  class="market"
>
  <h4
    class="market__title"
  >
    Player to Score First
  </h4>
  <div
    class="market__selections-wrapper"
  >
    <button
      class="market__btn "
      type="button"
    >
      <span>
        Messi
      </span>
      <span>
        1.3
      </span>
    </button>
  </div>
</section>
`;
